### ISSUE TITLE
Mappings to Basic Formal Ontology

### ISSUE THREAD

**Issue by nordlow** [ontologyportal/sumo#92]
The Basic Formal Ontology (BFO) contains an alternative partitioning of the upper ontology tree as follows (contents of [`bfo.txt`](https://github.com/BFO-ontology/BFO/blob/master/releases/2.0/bfo.txt)):

```
  entity (SUMO:Entity)
    occurrent
      process (SUMO:Process)
        process profile
        history
      temporal region
        one-dimensional temporal region
        zero-dimensional temporal region
      spatiotemporal region
      process boundary
    continuant (SUMO:Object)
      independent continuant
        immaterial entity (SUMO:Abstract)
          continuant fiat boundary
            one-dimensional continuant fiat boundary
            zero-dimensional continuant fiat boundary
            two-dimensional continuant fiat boundary
          spatial region
            three-dimensional spatial region
            two-dimensional spatial region
            one-dimensional spatial region
            zero-dimensional spatial region
          site
        material entity (SUMO:Physical)
          object aggregate
          object
          fiat object part
      generically dependent continuant
      specifically dependent continuant
        realizable entity
          disposition
            function
          role
        quality
          relational quality
```

As BFO is being considered to become an ISO-standard and is the upper ontology serving a great number of sub-ontologies at [Ontobee](http://www.ontobee.org/), I believe it would be a great idea to try to map/add/merge this taxonomy into SUMO's. Are there already any plans for doing this?

I've added SUMO-mappings inside parenthesises for the classes I found a one-to-one correspondence to. Please correct me if I made any mistakes.

BFO 2.0 doc is here https://github.com/BFO-ontology/BFO/blob/master/docs/bfo2-reference/BFO2-Reference.pdf.

**Comment by apease** [comment]
Hi Per,

   While I respect Barry Smith and have learned from conversation with 
him, and from reading his papers, I'm not sure what you mean by 
"advanced".  SUMO is much larger than BFO in its number of terms, and in 
its number of logical axioms defining each term.  I think you'll find 
that if you spend more time with SUMO that all the terms below are found 
in it, just by different names, and probably with a more detailed 
axiomatization, so there is nothing new there that needs integration.  
People do often have strong opinions about the aesthetics of different 
ontologies though and you should of course feel free to use ontologies 
other than SUMO if they are more to your liking.

all the best,
Adam

**Comment by nordlow** [comment]
I prefer SUMO over everything else I've seen, both with regards to file format (Lisp-like) and semantics.

I was mainly hoping that you could give me some guidance on mapping BFO to SUMO, so that I can connect SUMO with all the subontologies Ontobee.

**Comment by apease** [comment]
Hi Per,

   Many thanks for explaining.  A long time ago I tried mapping the OBO 
ontologies to SUMO but was frustrated because at least at the time they 
had, for the most part, only their term names to specify their meaning.  
So I couldn't map most terms since they had no definition.  I'm not 
familiar with Ontobee.  It would be great if there were more definitions 
there than in my previous experience.

   When there are actual definitions of terms however, the situation can 
also be quite frustrating, because my sense is there are many principled 
accounts of metaphysics.  When different theories of more concrete or 
specialized terms use a different supporting metaphysics, it's 
practically easier to start over from scratch.  So, in fact, that's my 
initial recommendation - if Ontobee has a useful inventory of terms, 
treat them like an informal glossary, rather than trying to convert 
their axiomatization.

   I took the briefest look at one term - anterior surface - 
http://www.ontobee.org/ontology/BSPO?iri=http://purl.obolibrary.org/obo/BSPO_0000371 
.  My suggestion would be to try to define that term as an extension to 
SUMO without respect to the existing hierarchy.  SUMO has notions of the 
back and front of an object and they both need further work (as well as 
&%surface, which is a bit better). But the Ontobee definition page has 
nothing other than simple links to related terms, and none of the 
axiomatization that would actually define, in a way suitable for 
automated inference, what that notion means.

   That's probably not a very satisfying recommendation, since it would 
be nice to envision connecting a few upper level terms and them being 
able to import thousands of lower level terms rendered compatible by 
that effort.  But I'm afraid that's not the case, at least as I see it.

   I would love to have your input however in taking a handful of terms 
in the most general terms sets in Ontobee, and working with you to show 
how they can be expressed in SUMO, in order to give you a head start on 
further axiomatization.


all the best,

Adam
